batch/batch/aioclient.py

```diff
@@ -105,25 +106,45 @@ def __init__(self, client, id, attributes):
 
 
 class BatchClient:
-    def __init__(self, session, url=None):
+    HAIL_NOTEBOOK_JWT_TOKEN_LOCATION = '/user-jwt/jwt'
+
+    def __init__(self, session, url=None, token_file=None, token=None, headers=None):
         if not url:
             url = 'http://batch.default'
         self.url = url
         self._session = session
+        if token is None:
+            token_file = (token_file or
+                          os.environ.get('HAIL_TOKEN_FILE') or
+                          os.path.expanduser('~/.hail/token'))
```

Comment:
Do we need so many locations for this credentials (HAIL_NOTEBOOK_JWT_TOKEN_LOCATION, in addition to all the others)?

Reply:
Hmm. We definitely need the explicit argument. The environment variable seems helpful to users. I suppose we could remove the `~/.hail/token`, but then users have to ensure the env var is set and I've found consistently setting an env var in all environments a little tricky. We could remove the notebook location, but then we'd have to set the env var or mount into `/home/jupyter/.hail/`. I think using a k8s projected volume to achieve the latter is pretty easy. Then we'd just have the explicit argument, env var, and the home dir location.

Reply:
Sure, then let’s remove the notebook location